src/explorer/TimelineExplorer.js

```diff
@@ -41,31 +42,38 @@ export class TimelineExplorer extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
     const timelineCred = props.initialTimelineCred;
-    const {alpha, intervalDecay, weights} = timelineCred.params();
+    const {alpha, intervalDecay} = timelineCred.params();
     this.state = {
       timelineCred,
       alpha,
       intervalDecay,
       // Set the weights to a copy, to ensure we don't mutate the weights in the
       // initialTimelineCred. This enables e.g. disabling the analyze button
       // when the parameters are unchanged.
-      weights: weightsCopy(weights),
+      weights: Weights.copy(timelineCred.weightedGraph().weights),
       loading: false,
       showWeightConfig: false,
       selectedNodeTypePrefix: null,
     };
   }
 
   params(): TimelineCredParameters {
-    const {alpha, intervalDecay, weights} = this.state;
+    const {alpha, intervalDecay} = this.state;
+    return {alpha, intervalDecay};
+  }
+
+  weights(): WeightsT {
     // Set the weights to a copy, to ensure that the weights we pass into e.g.
     // analyzeCred are a distinct reference from the one we keep in our state.
-    return {alpha, intervalDecay, weights: weightsCopy(weights)};
+    return Weights.copy(this.state.weights);
```

Comment:
Side-note: noticing we're probably copying the weights more than strictly needed.
But since that was the existing behavior, no change needed for this PR.

Reply:
I suspect we aren't. But happy to not change it. :)